### ISSUE TITLE
Loss dependent scheduler not supported (e.g. ReduceLROnPlateau)

### ISSUE THREAD

**Issue by bennyjg** [dreamquark-ai/tabnet#160]
**Describe the bug**

When using ReduceLROnPlateau scheduler, the scheduler expects input of epoch loss value to identify Plateau and raises error.
Adding stopping_loss as parameter to scheduler.step() in the code also does not help since it is located in the training section 
and not after the validation evaluation section.
I recommend as solution to move the scheduler.step lines to the Class TabModel section in the def of fit (line 244) and include stopping_loss (this time it is valid stopping_loss) as parameter.

**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce.**

**Expected behavior**

**Screenshots**

**Other relevant information:**
poetry version:  
python version: 
Operating System: 
Additional tools:

**Additional context**

**Comment by Optimox** [comment]
Hello,

Indeed at the moment loss dependent schedulers are note supported.

But I don't understand what you are saying after this, scheduler_fn is already present in TabModel and of course we are early stopping based on validation score not training score, otherwise it would not make sense at all.

I guess the easiest way of showing what you are thinking about would be to open a PR so that we can discuss the code directly!

**Comment by bennyjg** [comment]
Hi Optimox,
I hope the PR is clear. The solution was quit simple so I hope it will be adopted quickly.
Best,
Benny
